packages/@lwc/engine-core/src/framework/context-provider.ts

```diff
@@ -31,13 +31,13 @@ export function createContextProvider(adapter: WireAdapterConstructor) {
     }
     adapterContextToken = guid();
     setAdapterToken(adapter, adapterContextToken);
-    const providers: EventTarget[] = [];
+    const providers: WeakSet<EventTarget> = new WeakSet<EventTarget>();
```

Comment:
No need to set the types twice.

```suggestion
    const providers = new WeakSet<EventTarget>();
```